novawallet/Modules/Staking/NominationPools/BondMore/Setup/NominationPoolBondMorePresenter.swift

```diff
@@ -0,0 +1,182 @@
+        )
+    }
+
+    func provideTransferrableBalance() {
+        guard let balance = assetBalance?.transferable.decimal(precision: chainAsset.asset.precision) else {
+            view?.didReceiveTransferable(viewModel: nil)
+            return
+        }
+        let viewModel = balanceViewModelFactory.amountFromValue(balance).value(for: selectedLocale)
+        view?.didReceiveTransferable(viewModel: viewModel)
+    }
+
+    func provideAmountInputViewModel() {
+        guard let inputAmount = getInputAmount() else {
+            return
+        }
+        let viewModel = balanceViewModelFactory.createBalanceInputViewModel(
+            inputAmount
+        ).value(for: selectedLocale)
+
+        view?.didReceiveInput(viewModel: viewModel)
+    }
+
+    func provideAssetViewModel() {
+        let balance = transferrableBalance ?? 0
+        let inputAmount = getInputAmount() ?? 0
+        let viewModel = balanceViewModelFactory.createAssetBalanceViewModel(
+            inputAmount,
+            balance: balance,
+            priceData: price
+        ).value(for: selectedLocale)
+
+        view?.didReceiveAssetBalance(viewModel: viewModel)
+    }
+
+    override func getInputAmount() -> Decimal? {
+        guard let inputResult = inputResult else {
+            return nil
+        }
+        switch inputResult {
+        case .rate:
+            let fee = fee?.decimal(precision: chainAsset.asset.precision) ?? 0
+            let availableAmountDecimal = transferrableBalance ?? 0
+            return inputResult.absoluteValue(from: availableAmountDecimal - fee)
+        case let .absolute(inputValue):
+            return inputResult.absoluteValue(from: inputValue)
+        }
+    }
+
+    override func updateView() {
+        provideAssetViewModel()
+        provideAmountInputViewModel()
+        provideTransferrableBalance()
+        provideFee()
+        provideHints()
+    }
+
+    override func provideHints() {
+        let hints = hintsViewModelFactory.createHints(
+            rewards: claimableRewards,
+            locale: selectedLocale
+        )
+        view?.didReceiveHints(viewModel: hints)
+    }
+
+    override func provideFee() {
+        guard let fee = fee?.decimal(precision: chainAsset.asset.precision) else {
+            view?.didReceiveFee(viewModel: nil)
+            return
+        }
+
+        let viewModel = balanceViewModelFactory.balanceFromPrice(fee, priceData: price).value(for: selectedLocale)
+
+        view?.didReceiveFee(viewModel: viewModel)
+    }
+
+    override func getInputAmountInPlank() -> BigUInt? {
+        guard let decimalAmount = getInputAmount() else {
+            return nil
+        }
+
+        return decimalAmount.toSubstrateAmount(
+            precision: chainAsset.assetDisplayInfo.assetPrecision
+        )
+    }
+
+    override func didReceive(assetBalance: AssetBalance?) {
+        super.didReceive(assetBalance: assetBalance)
+
+        transferrableBalance = assetBalance.map {
```

Comment:
we can derive it from AssetBalance which we are also saving. No need to create a separate property

Reply:
We can derive decimal amount from AssetBalance too.